kubernetes_state/check.py

```diff
@@ -273,6 +297,32 @@ def kube_pod_status_phase(self, message, **kwargs):
             status = self.pod_phase_to_status.get(phase, self.UNKNOWN)
             self.service_check(check_basename + phase, status, tags=tags)
 
+    def kube_pod_container_status_waiting_reason(self, message, **kwargs):
+        metric_name = self.NAMESPACE + '.container.status_report.count.waiting'
+        for metric in message.metric:
+            tags = [self._format_tag(label.name, label.value) for label in metric.label]
```

Comment:
We should at least remove the `container` tag from these two metrics, see https://github.com/kubernetes/kube-state-metrics/blob/master/Documentation/pod-metrics.md

I'm still thinking even pod cardinality is overkill for that metric, I'd remove the `pod` label too, especially as we'll be submitting counts even on normal operations (terminated OK, waiting on containercreating), that'll spam a lot of contexts.
@hkaj can you chime in on that?

Reply:
Let's rename `container` in `kube_container_name` to be consistent with the `kubernetes` integration
Let's not submit the pod tag, and add a `//TODO` in agent6, use the tagger to get pod creator tags.

Reply:
`//todo` should also be here so users know that we purposefully skipped it but it's coming